test/unit/math/prim/mat/fun/gp_matern52_cov_test.cpp

```diff
@@ -0,0 +1,1398 @@
+               std::domain_error);
+
+  EXPECT_THROW(stan::math::gp_matern52_cov(x_bad_3, sigma, l_vec_bad),
+               std::domain_error);
+  EXPECT_THROW(stan::math::gp_matern52_cov(x_bad_3, sigma_bad, l_vec),
+               std::domain_error);
+  EXPECT_THROW(stan::math::gp_matern52_cov(x_bad_3, sigma_bad, l_vec_bad),
+               std::domain_error);
+}
+
+TEST(MathPrimMat, dim_mismatch_vec_eigen_rvec_gp_matern52_cov2) {
+  double sigma = 0.2;
+  double l = 5;
+
+  std::vector<Eigen::Matrix<double, 1, -1>> x_vec_1(3);
+  for (size_t i = 0; i < x_vec_1.size(); ++i) {
+    x_vec_1[i].resize(1, 3);
+    x_vec_1[i] << 1, 2, 3;
+  }
+
+  std::vector<Eigen::Matrix<double, 1, -1>> x_vec_2(4);
+  for (size_t i = 0; i < x_vec_2.size(); ++i) {
+    x_vec_2[i].resize(1, 4);
+    x_vec_2[i] << 4, 1, 3, 1;
+  }
+  EXPECT_THROW(stan::math::gp_matern52_cov(x_vec_1, x_vec_2, sigma, l),
+               std::invalid_argument);
+}
+
+TEST(MathPrimMat, dim_mismatch_vec_eigen_mixed_gp_matern52_cov) {
+  double sigma = 0.2;
+  double l = 5;
+
+  std::vector<Eigen::Matrix<double, 1, -1>> x_rvec_1(3);
+  for (size_t i = 0; i < x_rvec_1.size(); ++i) {
+    x_rvec_1[i].resize(1, 3);
+    x_rvec_1[i] << 1, 2, 3;
+  }
+
+  std::vector<Eigen::Matrix<double, -1, 1>> x_vec_1(4);
+  for (size_t i = 0; i < x_vec_1.size(); ++i) {
+    x_vec_1[i].resize(4, 1);
+    x_vec_1[i] << 4, 1, 3, 1;
+  }
+
+  std::vector<Eigen::Matrix<double, -1, 1>> x_vec_2(3);
+  for (size_t i = 0; i < x_vec_2.size(); ++i) {
+    x_vec_2[i].resize(3, 1);
+    x_vec_2[i] << 1, 2, 3;
+  }
+
+  std::vector<Eigen::Matrix<double, 1, -1>> x_rvec_2(4);
+  for (size_t i = 0; i < x_rvec_2.size(); ++i) {
+    x_rvec_2[i].resize(1, 4);
+    x_rvec_2[i] << 1, 2, 3, 4;
+  }
+  EXPECT_THROW(stan::math::gp_matern52_cov(x_rvec_1, x_vec_1, sigma, l),
+               std::invalid_argument);
+  EXPECT_THROW(stan::math::gp_matern52_cov(x_vec_1, x_rvec_1, sigma, l),
+               std::invalid_argument);
+  EXPECT_THROW(stan::math::gp_matern52_cov(x_rvec_2, x_vec_2, sigma, l),
+               std::invalid_argument);
+  EXPECT_THROW(stan::math::gp_matern52_cov(x_vec_2, x_rvec_2, sigma, l),
+               std::invalid_argument);
+}
+
+TEST(MathPrimMat, ard_size_err_gp_matern52_cov) {
+  double sigma = 0.2;
+
+  std::vector<double> l(7);
+  for (int i = 0; i < 7; ++i)
+    l[i] = i + 1;
+
+  std::vector<Eigen::Matrix<double, -1, 1>> x_vec_1(3);
+  for (size_t i = 0; i < x_vec_1.size(); ++i) {
+    x_vec_1[i].resize(3, 1);
+    x_vec_1[i] << 1, 2, 3;
+  }
+
+  std::vector<Eigen::Matrix<double, -1, 1>> x_vec_2(3);
+  for (size_t i = 0; i < x_vec_2.size(); ++i) {
+    x_vec_2[i].resize(3, 1);
+    x_vec_2[i] << 1, 2, 3;
+  }
+
+  EXPECT_THROW(stan::math::gp_matern52_cov(x_vec_1, sigma, l),
+               std::invalid_argument);
+  EXPECT_THROW(stan::math::gp_matern52_cov(x_vec_1, x_vec_2, sigma, l),
+               std::invalid_argument);
+}
+
+TEST(MathPrimMat, blahblah) {
```

Comment:
`blahblah`? I don't think this is meant to be here based on the content of the test. Please remove if not useful.